tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -662,84 +862,329 @@ TEST(RDFVary, VaryDisplay) // TEST instead of TEST_P because Display is single-t
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetHisto().GetMean(), 43.);
 }
 
-TEST_P(RDFVary, DISABLED_VaryHisto3D)
+TEST_P(RDFVary, VaryGraph)
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Graph<int, int>("x", "x");
+   auto hs = VariationsFor(h);
 
+   EXPECT_DOUBLE_EQ(h->GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMean(), 3.5);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMean(), 5.5);
 }
 
-TEST_P(RDFVary, DISABLED_VaryHistoND)
+TEST_P(RDFVary, VaryGraphAsymmErrors)
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .GraphAsymmErrors<int, int, int, int, int, int>("x", "x", "x", "x", "x", "x");
+   auto hs = VariationsFor(h);
 
-}
-
-TEST_P(RDFVary, DISABLED_VaryMax)
-{
+   EXPECT_DOUBLE_EQ(h->GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(hs["nominal"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(hs["x:0"].GetMean(), 3.5);
+   EXPECT_DOUBLE_EQ(hs["x:1"].GetMean(), 5.5);
+}
+
+TEST_P(RDFVary, VaryHistos)
+{
+   auto df = ROOT::RDataFrame(10)
+                .Define("x",
+                        [] {
+                           return ROOT::RVecI{1, 2, 3};
+                        })
+                .Vary(
+                   "x",
+                   [] {
+                      return ROOT::RVec<ROOT::RVecI>{{}, {4, 5}};
+                   },
+                   {}, 2);
+   auto h1 = df.Histo1D<ROOT::RVecI>("x");
+   auto h1s = VariationsFor(h1);
+   auto h2 = df.Histo2D<ROOT::RVecI, ROOT::RVecI>({}, "x", "x");
+   auto h2s = VariationsFor(h2);
+   auto h3 = df.Histo3D<ROOT::RVecI, ROOT::RVecI, ROOT::RVecI>({}, "x", "x", "x");
+   auto h3s = VariationsFor(h3);
+
+   EXPECT_DOUBLE_EQ(h1->GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2->GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h3->GetMean(), 2.);
+
+   EXPECT_DOUBLE_EQ(h1s["nominal"].GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h2s["nominal"].GetMean(), 2.);
+   EXPECT_DOUBLE_EQ(h3s["nominal"].GetMean(), 2.);
+
+   EXPECT_DOUBLE_EQ(h1s["x:0"].GetMean(), 0.);
+   EXPECT_DOUBLE_EQ(h2s["x:0"].GetMean(), 0.);
+   EXPECT_DOUBLE_EQ(h3s["x:0"].GetMean(), 0.);
+
+   EXPECT_DOUBLE_EQ(h1s["x:1"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(h2s["x:1"].GetMean(), 4.5);
+   EXPECT_DOUBLE_EQ(h3s["x:1"].GetMean(), 4.5);
+
+   int nbins[4] = {10, 10, 10, 10};
+   double xmin[4] = {0., 0., 0., 0.};
+   double xmax[4] = {100., 100., 100., 100.};
+   auto hN = df.HistoND<ROOT::RVecI, ROOT::RVecI, ROOT::RVecI, ROOT::RVecI>({"name", "title", 4, nbins, xmin, xmax},
```

Comment:
```suggestion
   auto hN = df.HistoND<ROOT::RVecI, ROOT::RVecI, ROOT::RVecI, ROOT::RVecI>({"", "", 4, nbins, xmin, xmax},
```

This should avoid some warnings about histograms with the same name being registered with ROOT multiple times. Similarly for other tests below.

Reply:
I applied this change, but I still cannot remove the warning:
```
Warning in <TROOT::Append>: Replacing existing TH1: _proj_3 (Potential memory leak).
```
coming from: `EXPECT_DOUBLE_EQ(hNs["nominal"].Projection(3)->GetMean(), 5.);`